lib/improver/utilities/statistical_operations.py

```diff
@@ -0,0 +1,427 @@
+        distribution to obtain a probability. The point to which we interpolate
+        is defined by the threshold_cube.
+
+        Note that the current implementation assumes that in cases of a
+        degenerate percentile distribution, the right most bin in which a
+        threshold value is found is chosen.
+
+        e.g.::
+
+            Percentile: 0 10 20 30 40 50 ...
+            Height (m): 0 0 0 15 30 40 ...
+
+            A height of 0m will be associated with a probabilty of 20%. This is
+            not correct, but nor is the approach of taking 0%. The percentile
+            approach is not suitable with these degenerate distributions, so be
+            wary of the returned probabilities.
+
+        Examples:
+
+            This simple linear interpolator works in the following way.
+
+            percentiles_cube::
+
+                [ [[2.0, 2.0, 2.0],
+                   [2.0, 2.0, 2.0],
+                   [2.0, 2.0, 2.0]],
+
+                  [[4.0, 4.0, 4.0],
+                   [4.0, 4.0, 4.0],
+                   [4.0, 4.0, 4.0]] ]
+
+            threshold_cube::
+
+                [ [1.0, 1.0, 1.0],
+                  [3.0, 3.0, 3.0],
+                  [5.0, 5.0, 5.0] ]
+
+            value_bounds::
+
+                [ [[np.nan, np.nan, np.nan],
+                   [np.nan, np.nan, np.nan],
+                   [np.nan, np.nan, np.nan]],
+
+                  [[np.nan, np.nan, np.nan],
+                   [np.nan, np.nan, np.nan],
+                   [np.nan, np.nan, np.nan]] ]
+
+            percentile_bounds::
+
+                [ [[-1, -1, -1],
+                   [-1, -1, -1],
+                   [-1, -1, -1]],
+
+                  [[-1, -1, -1],
+                   [-1, -1, -1],
+                   [-1, -1, -1]] ]
+
+
+            1. Create slices over each percentile, and using the correct
+               inequality (as determined by inverse_ordering) compare the
+               threshold values to the percentiles slice; here we assume
+               inverse_ordering is False, so we use >=. We then populate the
+               value_bounds and percentile_bounds arrays.
+
+               Slice 0 - 0th Percentile::
+
+                   [[1.0 >= 2.0, 1.0 >= 2.0, 1.0 >= 2.0],
+                    [3.0 >= 2.0, 3.0 >= 2.0, 3.0 >= 2.0],
+                    [5.0 >= 2.0, 5.0 >= 2.0, 5.0 >= 2.0]]
+
+                   [[False, False, False],
+                    [True, True, True],
+                    [True, True, True]]
+
+               The value_bounds array has a leading dimensions with 2 indices
+               to be associated with the lower [0] and upper bounds [1] about
+               the threshold being considered. The [0] index is populated with
+               the values in the slice of percentiles_cube at every True index.
+               The [1] index is populated with the values in the next slice of
+               percentiles_cube.::
```

Comment:
This does seem to show up as 
`.:` in the documentation followed by the code block.

Reply:
✅